nautobot/dcim/forms.py

```diff
@@ -3235,7 +3235,9 @@ def __init__(self, *args, **kwargs):
             # Limit VLAN choices by Location
             if locations.count() == 1:
                 location = locations.first()
-                self.fields["untagged_vlan"].widget.add_query_param("locations", location.pk)
+                # In the case of a single location, use the available_on_device query param to limit untagged VLAN choices
+                # to those available on the devices in that location and in the ancestors of the location.
+                self.fields["untagged_vlan"].widget.add_query_param("available_on_device", device.pk)
```

Comment:
I'm confused now, don't we still need to remove `locations: null` from the base field definition (line 3187) for this to work?